lib/server.js

```diff
@@ -141,18 +141,25 @@ module.exports = function createServer(config, log) {
     // that we won't block. These are typically for Mozilla QA purposes.
     // They should be checked after everything else so as not to pay the
     // overhead of checking the many requests that are *not* QA-related.
-    if (result.block || result.suspect) {
-      if (ip in allowedIPs.ips || (email && allowedEmailDomains.isAllowed(email))) {
-        log.info({
-          op: 'request.check.allowed',
-          ip: ip,
-          block: result.block,
-          suspect: result.suspect
-        })
-        result.block = false
-        result.suspect = false
-      }
+    if ((result.block || result.suspect) && isWhitelisted(ip, email)) {
+      log.info({
+        op: 'request.check.allowed',
+        ip: ip,
+        block: result.block,
+        suspect: result.suspect
+      })
+
+      result.block = false
+      result.suspect = false
+
+      return true
     }
+
+    return false
+  }
+
+  function isWhitelisted (ip, email) {
```

Comment:
FWIW, a while ago we invested a bit of effort into replacing "whitelisted" with other adjectives like "trusted" or "allowlisted" in service of inclusivity; let's maintain that maintain that state unless we have a good reason not to.